lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -155,14 +155,18 @@ instance NFData (Wallet s t) where
                           Construction & Modification
 -------------------------------------------------------------------------------}
 
--- | Create an empty wallet from an initial state
+-- | Create a an empty wallet and apply the given genesis block
+--
+-- The wallet tip will be set to the header of the applied genesis block.
 initWallet
-    :: (IsOurs s, NFData s, Show s, TxId t)
-    => BlockHeader
-    -- ^ Very first 'BlockHeader'
+    :: forall s t. (IsOurs s, NFData s, Show s, TxId t)
+    => Block
+    -- ^ The genesis block
     -> s
     -> Wallet s t
-initWallet = Wallet mempty mempty
+initWallet block s =
+    let ((txs, utxo'), s') = prefilterBlock (Proxy @t) block mempty s
+    in Wallet utxo' (Set.fromList txs) (header block) s'
```

Comment:
Hmmmm. What about `applyBlock block (Wallet mempty mempty (header block) s`  ? 
Feels a bit weird to copy paste partial stuff from `applyBlock`, especially because, there _could_ be some transactions that are ours in the genesis block (and there will be actually).

Reply:
We will then transition from
`(Wallet mempty mempty (header block) s`
to
`(Wallet x y (header block) s'`

i.e, the header inside the first `applyBlock` will be incorrect.

I'd be fine with this if it indeed compiles though.

Reply:
Hmmm. It's probably not important for now since there will be no transactions yet. And maybe it's best to leave it for later actually until we have confirmation of _how_ the current mainnet is going to be included in the genesis block. So, leave it be :+1:

Reply:
> could be some transactions that are ours in the genesis block

I thought `(Set.fromList txs)` would add them though 🤔 ?